engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageFunctionUtils.java

```diff
@@ -19428,4 +19430,436 @@ public static int negate(short a) {
         return a == QueryConstants.NULL_SHORT ? QueryConstants.NULL_INT : -a;
     }
 
+    public static BigDecimal plus(BigDecimal a, long b) {
+        if (a == null || b == QueryConstants.NULL_LONG) {
+            return null;
+        }
+        return a.add(BigDecimal.valueOf(b));
+    }
+
+    public static BigDecimal plus(long a, BigDecimal b) {
+        if (a == QueryConstants.NULL_LONG || b == null) {
+            return null;
+        }
+        return plus(b, a);
+    }
+
+    public static BigDecimal plus(BigDecimal a, double b) {
+        if (a == null || b == QueryConstants.NULL_DOUBLE) {
+            return null;
+        }
+        return a.add(BigDecimal.valueOf(b));
+    }
+
+    public static BigDecimal plus(double a, BigDecimal b) {
+        if (a == QueryConstants.NULL_DOUBLE || b == null) {
+            return null;
+        }
+        return plus(b, a);
+    }
+
+    public static BigDecimal minus(BigDecimal a, long b) {
+        if (a == null || b == QueryConstants.NULL_LONG) {
+            return null;
+        }
+
+        return a.subtract(BigDecimal.valueOf(b));
+    }
+
+    public static BigDecimal minus(long a, BigDecimal b) {
+        if (a == QueryConstants.NULL_LONG || b == null) {
+            return null;
+        }
+        return BigDecimal.valueOf(a).subtract(b);
+    }
+
+    public static BigDecimal minus(BigDecimal a, double b) {
+        if (a == null || b == QueryConstants.NULL_DOUBLE) {
+            return null;
+        }
+        return a.subtract(BigDecimal.valueOf(b));
+    }
+
+    public static BigDecimal minus(double a, BigDecimal b) {
+        if (a == QueryConstants.NULL_DOUBLE || b == null) {
+            return null;
+        }
+        return BigDecimal.valueOf(a).subtract(b);
+    }
+
+    public static BigDecimal multiply(BigDecimal a, long b) {
+        if (a == null || b == QueryConstants.NULL_LONG) {
+            return null;
+        }
+        return a.multiply(BigDecimal.valueOf(b));
+    }
+
+    public static BigDecimal multiply(long a, BigDecimal b) {
+        if (a == QueryConstants.NULL_LONG || b == null) {
```

Comment:
should you really repeat checks if you are just going to use the commutative operator?